### DISCUSSION TITLE
Does flax/jax care about image channel order?

### DISCUSSION THREAD

**Discussion by davidshen84** [google/flax#3501] [Q&A] [answered]
Hi,

Most ML libraries prefer images to be pre-processed to be *channel first*. But I could not find a reference about how flax/jax prefer it.

Does flax take care of it automatically? (*I don't think it is possible*)

**Comment by chiamp** [comment, marked as answer]
For the `Conv` layer, it's [_channels-last_](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/_autosummary/flax.linen.Conv.html#flax.linen.Conv.__call__) convention

**Reply by epignatelli** [reply]
That's the correct answer, the one marked  as "answer" is incorrect at the date of this comment.
See the [docs](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/layers.html#flax.linen.Conv.__call__)

**Comment by davidshen84** [comment]
Thanks.

I noticed the underlying implementation uses
https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.conv_general_dilated.html,
which defaults channel-first. Is there any reason we changed the convention
in flax?

**Reply by chiamp** [reply]
perhaps @marcvanzee would know

**Comment by ChandraPrakash-Bathula** [comment]


**Reply by epignatelli** [reply]
Where do you specify which convention to choose?

**Reply by epignatelli** [reply]
In the docs I read:
> inputs – input data with dimensions ([*](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/layers.html#id3)batch_dims, spatial_dims…, features). This is the **channels-last** convention, i.e. NHWC for a 2d convolution and NDHWC for a 3D convolution.
